sequence/fasta_io.hpp

```diff
@@ -6,25 +6,39 @@
 #include <filesystem>
 #include <fstream>
 #include <iostream>
-#include <string>
+#include <iterator>
 #include <sstream>
+#include <string>
 
 namespace ts { // ts = Tensor Sketch
 
+// Represents the contents of a single Fasta file.
+// All the sequences in a file should be treated as a single assembly and should be sketched as a
+// whole.
+template <typename seq_type>
+struct FastaFile {
+    // The name of the file
+    std::string filename;
+    // The leading comment before each sequence. Always has the same length as sequences.
+    std::vector<std::string> comments;
+    // The sequences in the file
+    std::vector<std::vector<seq_type>> sequences;
```

Comment:
We typically use Doxygen-style comments (`/** */`) when commenting on classes, members, methods and `//` is reserved for comments in the code